### PR TITLE
Umap polish

### PR DESCRIPTION
**Pull request by jashapiro**
### Purpose

As part of the ongoing polishing for #289 & #290, here I am adding edits to the two UMAP notebooks. 

I don't think there are major change here, aside from bringing some of the commentary from the RNA-seq example back to the microarray.

Most of the other changes are translations of the changes that came with PCA

#### Gotchas the reviewer should know about

Please do look at the rendered files in case there are things there that I missed.

## Remaining concerns and questions

These do keep the filtering after making DEseq objects: we still might want to filter first as described in #417

### CODE REVIEW

File: 02-microarray/dimension-reduction_microarray_02_umap.Rmd [hunk 1]
```diff
@@ -216,50 +216,55 @@ metadata <- readr::read_tsv(metadata_file)
...
+# Perform UMAP using the `umap::umap()` function
 umap_data <- umap::umap(
-  t(df) # We have to transpose our data frame so we are obtaining UMAP scores for samples instead of genes
+  t(df) # transpose our data frame to obtain PC scores for samples, not genes
 )
 
 # Make into data frame for plotting with `ggplot2`
-umap_df <- data.frame(umap_data$layout) %>% # The umap values we need for plotting are stored in this `layout` element
-  # Turn samples_ids stored as rownames into column
+# The umap values we need for plotting are stored in the `layout` element
```
Comment (cansavvy) [inline 02-microarray/dimension-reduction_microarray_02_umap.Rmd:248]:
```suggestion
# The UMAP values we need for plotting are stored in the `layout` element
```

Just style reasons.

File: 02-microarray/dimension-reduction_microarray_02_umap.Rmd [hunk 2]
```diff
@@ -324,17 +329,30 @@ umap_plot
 Our histological subtype groups don't appear to be clustering in a discernible pattern. 
 We could test out other variables as annotation labels to get a further understanding of the cluster behavior of each subgroup.
 
+### Interpretation of UMAP plot and results
+
+1. Note that the coordinates of UMAP output for any given cell can change dramatically depending on parameters, and even run to run with the same parameters  (Also why setting the seed is important).
```
Comment (cansavvy) [inline 02-microarray/dimension-reduction_microarray_02_umap.Rmd:334]:
```suggestion
1. Note that the coordinates of UMAP output for any given cell can change dramatically depending on parameters, even if ran with the same parameters (Also why setting the seed is important).
<!-- TODO: set the seed glossary link --> 
```

File: 03-rnaseq/dimension-reduction_rnaseq_02_umap.Rmd [hunk 3]
```diff
@@ -337,12 +340,24 @@ umap_results <- umap::umap(normalized_counts)
 Now that we have the results from UMAP, we need to extract the counts data from the `umap_results` object and merge the variables from the metadata that we will use for annotating our plot.
 
 ```{r}
-# We need to tell R to create a data frame with the umap values and annotation data in preparation for plotting with `ggplot2`
+# Make into data frame for plotting with `ggplot2`
+# The umap values we need for plotting are stored in the `layout` element
```
Comment (cansavvy) [inline 03-rnaseq/dimension-reduction_rnaseq_02_umap.Rmd:344]:
```suggestion
# The UMAP values we need for plotting are stored in the `layout` element
```

File: 03-rnaseq/dimension-reduction_rnaseq_02_umap.Rmd [hunk 4]
```diff
@@ -337,12 +340,24 @@ umap_results <- umap::umap(normalized_counts)
...
 umap_plot_df <- data.frame(umap_results$layout) %>%
-  tibble::rownames_to_column("refinebio_accession_code") %>% # Let's get the rownames into a column so we can merge the annotation data
-  dplyr::inner_join(metadata, by = "refinebio_accession_code") # Let's merge the annotation data using the `refinebio_accession_code` column
+  # Turn sample IDs stored as row names into column
```
Comment (cansavvy) [inline 03-rnaseq/dimension-reduction_rnaseq_02_umap.Rmd:346]:
```suggestion
  # Turn sample IDs stored as row names into a column
```

File: 02-microarray/dimension-reduction_microarray_02_umap.Rmd [hunk 5]
```diff
@@ -216,50 +216,55 @@ metadata <- readr::read_tsv(metadata_file)
...
-  t(df) # We have to transpose our data frame so we are obtaining UMAP scores for samples instead of genes
+  t(df) # transpose our data frame to obtain PC scores for samples, not genes
 )
 
 # Make into data frame for plotting with `ggplot2`
-umap_df <- data.frame(umap_data$layout) %>% # The umap values we need for plotting are stored in this `layout` element
-  # Turn samples_ids stored as rownames into column
+# The umap values we need for plotting are stored in the `layout` element
+umap_df <- data.frame(umap_data$layout) %>%
+  # Turn sample IDs stored as row names into column
```
Comment (cansavvy) [inline 02-microarray/dimension-reduction_microarray_02_umap.Rmd:250]:
```suggestion
  # Turn sample IDs stored as row names into a column
```

File: 02-microarray/dimension-reduction_microarray_02_umap.Rmd [hunk 6]
```diff
@@ -216,50 +216,55 @@ metadata <- readr::read_tsv(metadata_file)
...
-  # Tuck away the gene ID  column as rownames
+  # Tuck away the gene ID  column as rownames, leaving only numeric values
   tibble::column_to_rownames("Gene")
 ```
 
 Let's ensure that the metadata and data are in the same sample order. 
 
 ```{r}
-# Make the data in the order of the metadata
+# Make the sure the columns (samples) are in the same order as the metadata
```
Comment (cansavvy) [inline 02-microarray/dimension-reduction_microarray_02_umap.Rmd:226]:
```suggestion
# Make sure the columns (samples) are in the same order as the metadata
```

File: 02-microarray/dimension-reduction_microarray_02_umap.Rmd [hunk 7]
```diff
@@ -324,17 +329,30 @@ umap_plot
 Our histological subtype groups don't appear to be clustering in a discernible pattern. 
 We could test out other variables as annotation labels to get a further understanding of the cluster behavior of each subgroup.
 
+### Interpretation of UMAP plot and results
```
Comment (cansavvy) [inline 02-microarray/dimension-reduction_microarray_02_umap.Rmd:332]: Did you not want to also add this section to the RNA-seq version? (or was it already there and I can't see it)

**Review by cansavvy — COMMENTED** [review]
I think these polishing items look good! I just have tiny comments.

File: 02-microarray/dimension-reduction_microarray_02_umap.Rmd (hunk 7 above)
Comment (jashapiro) [reply 02-microarray/dimension-reduction_microarray_02_umap.Rmd:332]: Yup, it is already there: copied it directly!

**Review by cansavvy — APPROVED** [review]
Looks great to me!

**Pull request merged**